src/probnum/problems/_problems.py

```diff
@@ -1 +1,204 @@
+    0.09
+    """
+
+    f: typing.Callable[[float, np.ndarray], np.ndarray]
+    t0: float
+    tmax: float  # Bold move: remove this? This is not really part of an IVP.
+    y0: typing.Union[pntp.FloatArgType, np.ndarray]
+    df: typing.Optional[typing.Callable[[float, np.ndarray], np.ndarray]] = None
+    ddf: typing.Optional[typing.Callable[[float, np.ndarray], np.ndarray]] = None
+
+    # For testing and benchmarking
+    solution: typing.Optional[typing.Callable[[float, np.ndarray], np.ndarray]] = None
+
+
+@dataclasses.dataclass
+class LinearSystem:
+    r"""Linear system of equations.
+
+    Compute :math:`x` from :math:`Ax=b`.
+    Solved by the probabilistic linear solver in :mod:`probnum.linalg`
+
+    Parameters
+    ----------
+    A
+        Square matrix or linear operator.
+    b
+        Right-hand side vector or matrix.
+    solution
+        Closed form, analytic solution to the problem. Used for testing and benchmarking.
+
+    Examples
+    --------
+    >>> A = np.eye(3)
+    >>> b = np.arange(3)
+    >>> lin_sys = LinearSystem(A, b)
+    >>> lin_sys
+    LinearSystem(A=array([[1., 0., 0.],
+           [0., 1., 0.],
+           [0., 0., 1.]]), b=array([0, 1, 2]), solution=None)
+    """
+
+    A: typing.Union[
+        np.ndarray,
+        scipy.sparse.spmatrix,
+        pnlo.LinearOperator,
+        pnrv.RandomVariable,
+    ]
+    b: typing.Union[np.ndarray, pnrv.RandomVariable]
+
+    # For testing and benchmarking
+    solution: typing.Optional[typing.Union[np.ndarray, pnrv.RandomVariable]] = None
+
+
+@dataclasses.dataclass
+class QuadratureProblem:
+    r"""Numerical computation of an integral.
+
+    Compute the integral
+
+        .. math::
+            \int_\Omega f(x) \, \text{d} \mu(x)
+
+    for a function :math:`f: \Omega \rightarrow \mathbb{R}`.
+    For the time being, :math:`\mu` is the Lebesgue measure.
+    Solved by the quadrature rules in :mod:`probnum.quad`.
+
+    Parameters
+    ----------
+    integrand
+        Function to be integrated.
+    lower_bd
+        A number or a vector representing the lower bounds of the integrals.
+    upper_bd
+        A number or a vector representing the upper bounds of the integrals.
+    output_dim
+        Output dimension of the integrand.
+    solution
+        Closed form, analytic solution to the problem. Used for testing and benchmarking.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> def integrand(x):
+    ...     return np.linalg.norm(x)**2
+    >>> lower_bd = 0.41
+    >>> upper_bd = 4.32
+    >>> qp1d = QuadratureProblem(integrand, lower_bd=lower_bd, upper_bd=upper_bd)
+    >>> np.round(qp1d.integrand(0.2), 2)
+    0.04
+    >>> qp1d.lower_bd
+    0.41
+    >>>
+    >>> lower_bd = [0., 0.]
+    >>> upper_bd = [1., 1.]
+    >>> qp2d = QuadratureProblem(integrand, lower_bd=lower_bd, upper_bd=upper_bd)
+    >>> qp2d.upper_bd
+    [1.0, 1.0]
+    """
+
+    integrand: typing.Callable[[np.ndarray], typing.Union[float, np.ndarray]]
```

Comment:
f could also be a `RandomProcess`?

Reply:
If that is something you intend to cover with BQ, sure! We can update the type hint as soon as we have random processes!

Reply:
Maybe leave the `RandomProcess` part out for now, still can add this later